### PR TITLE
feat: streaming for standard brain picking

### PR DESCRIPTION
**Pull request by mattzcarey**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

Standard brain picking (BE)
Frontend

TODO: OpenAI function future PR

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by StanGirard** [comment]
image

**Review by StanGirard — COMMENTED** [review]
remove brainpicking.cpython

File: backend/vectorstore/supabase.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def __init__(
     def similarity_search(
         self,
         query: str,
+        user_id: str = "none",
```
Comment (StanGirard) [inline backend/vectorstore/supabase.py:28]: why ?

**Review by StanGirard — CHANGES_REQUESTED** [review]
- Remove cpython file
- Explain or remove supabase.py str for user_id.

Else looks good <3 
Test and build work

File: backend/vectorstore/supabase.py (hunk 1 above)
Comment (mattzcarey) [reply backend/vectorstore/supabase.py:28]: This was the bug we found no? is it not relevant any more?

Comment (mattzcarey) [reply backend/vectorstore/supabase.py]: This was the big we previously had with the documents? I'll check it out today anyway.

**Review by StanGirard — APPROVED** [review]

File: frontend/app/chat/[chatId]/hooks/useChat.ts [hunk 2]
```diff
@@ -21,30 +22,34 @@ export const useChat = () => {
...
   } = useChatService();
 
   useEffect(() => {
     const fetchHistory = async () => {
-      const chatHistory = await getChatHistory(chatId);
-      setHistory(chatHistory);
+      const currentChatId = chatId;
+      const chatHistory = await getChatHistory(currentChatId);
+
+      if (chatId === currentChatId && chatHistory.length > 0) {
```
Comment (mamadoudicko) [inline frontend/app/chat/[chatId]/hooks/useChat.ts:40]: Not sure to understand what is happening here:
```
const currentChatId = chatId;
if(chatId === currentChatId)
````

File: frontend/lib/helpers/uuid.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export const generateUUID = (): string => {
```
Comment (mamadoudicko) [inline frontend/lib/helpers/uuid.ts:1]: Can now be removed i think

**Review by mamadoudicko — APPROVED** [review]
Nice job @mattzcarey !

Few low Loggaf returns but LGFM!

File: frontend/app/chat/[chatId]/hooks/useChat.ts (hunk 2 above)
Comment (mattzcarey) [reply frontend/app/chat/[chatId]/hooks/useChat.ts:40]: fair :) that's dead. I think I didnt spot the line above.

File: frontend/lib/helpers/uuid.ts (hunk 3 above)
Comment (mattzcarey) [reply frontend/lib/helpers/uuid.ts:1]: yes definitely.

**Pull request merged**